ambry-api/src/main/java/com/github/ambry/router/Router.java

```diff
@@ -88,6 +88,22 @@
    */
   Future<Void> updateBlobTtl(String blobId, String serviceId, long expiresAtMs, Callback<Void> callback);
 
+  /**
+   * Requests for a blob to be undeleted asynchronously and invokes the {@link Callback} when the request completes.
+   * @param blobId The ID of the blob that needs to be undeleted.
+   * @param serviceId The service ID of the service undeleting the blob. This can be null if unknown.
+   * @param callback The {@link Callback} which will be invoked on the completion of a request.
+   * @return A future that would contain information about whether the undelete succeeded or not, eventually.
+   */
+  default Future<Void> undeleteBlob(String blobId, String serviceId, Callback<Void> callback) {
```

Comment:
Based on my recent experience, `default` may cause some test issues because it brings in a default implementation. Prefer to have same interface as `updateBlobTtl `.

Reply:
I can remove it.